src/harness/fourslash.ts

```diff
@@ -1797,6 +1797,10 @@ namespace FourSlash {
                 // this.languageService.getScriptLexicalStructure(fileName);
             }
 
+            if (this.currentCaretPosition < 0) {
+                this.currentCaretPosition = 0;

```

Comment:
> If the caret is in the middle of some range that is deleted, should the caret go to the beginning of the range?

@aozgaa yes. This was also the conclusion we arrived with @amcasey after discussing offline. Will make that change.
Perfectly valid scenario - this happens in remove unused quickfix, for instance (which is I guess why I was hitting the invalid caret situation)